src/G2/Equiv/Induction.hs

```diff
@@ -0,0 +1,336 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module G2.Equiv.Induction
+    ( inductionFull
+    , prevFiltered
+    )
+    where
+
+-- TODO may not need all imports
+
+import G2.Language
+
+import G2.Config
+
+import G2.Interface
+
+import qualified Control.Monad.State.Lazy as CM
+
+import qualified G2.Language.ExprEnv as E
+import qualified G2.Language.Typing as T
+import qualified G2.Language.CallGraph as G
+
+import Data.List
+import Data.Maybe
+import qualified Data.Text as DT
+
+import qualified Data.HashSet as HS
+import qualified G2.Solver as S
+
+import qualified G2.Language.PathConds as P
+
+import G2.Equiv.InitRewrite
+import G2.Equiv.EquivADT
+import G2.Equiv.G2Calls
+import G2.Equiv.Tactics
+
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.Map as M
+import G2.Execution.Memory
+import Data.Monoid (Any (..))
+
+import Debug.Trace
+
+import G2.Execution.NormalForms
+import qualified G2.Language.Stack as Stck
+import Control.Monad
+
+import Data.Time
+
+import G2.Execution.Reducer
+import G2.Lib.Printers
+
+import qualified Control.Monad.Writer.Lazy as W
+
+swap :: (a, b) -> (b, a)
+swap (x, y) = (y, x)
```

Comment:
This already exists in Data.Tuple:
https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Tuple.html#v:swap